### PR TITLE
S8 Code refactor (1)

### PR DESCRIPTION
**Pull request by DanThePol**
- Made Repositories static objects and updated viewmodels to reflect this change
- Removed useless static elements from constructors (like Firebase.firestore)
- Simplified other function signatures when possible (e.g. from a bunch of fields from Event to just passing the event as a parameter)
- Converted all the Spacers with hardcoded size I found to screen size dependent Spacers
- Moved certain code snippets to private helper functions and added/redefined some enums for code legibility (for instance in Events.kt)
- Moved certain helper functions into the model folder (for example getCountries went from ui.authscreens.register.RegisterNameCountryPhone.kt to model.authentication.SignInState.kt)
- Added missing documentation

This is a massive PR, I should've broken this down into smaller PRs. More code refactoring to come, I'm still not done with the main.java.com.github.se.gomeet folder.

### CODE REVIEW

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/ManageInvites.kt
Comment (eloibressaud) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/ManageInvites.kt:1]: Way clearer using enum and separate helper functions.

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/Events.kt
Comment (eloibressaud) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/Events.kt:1]: Again this is way clearer, well done.

File: app/src/main/java/com/github/se/gomeet/ui/theme/Theme.kt
Comment (eloibressaud) [inline app/src/main/java/com/github/se/gomeet/ui/theme/Theme.kt:1]: I believe onBackground = Color.Black should stay even if not really necessary for clarity reason.

**Review by eloibressaud — CHANGES_REQUESTED** [review]
This should have been multiple PRs as it touches on multiple parts of the code in various ways, making it hard to follow.

It however greatly improve on code readability and is an overall great improvement. As it passes all the automated tests and holds up to basic manual testing i believe it should be merged once the small issue with Theme.kt has been solved.

File: app/src/main/java/com/github/se/gomeet/ui/theme/Theme.kt
Comment (DanThePol) [reply app/src/main/java/com/github/se/gomeet/ui/theme/Theme.kt:1]: Well spotted, I just added it back!

**Review by eloibressaud — APPROVED** [review]
Changes solved the previous issue
